crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -27,12 +27,20 @@ pub struct FileAssetIo {
 }
 
 impl FileAssetIo {
-    pub fn new<P: AsRef<Path>>(path: P) -> Self {
-        FileAssetIo {
+    pub fn new<P: AsRef<Path>>(
+        path: P,
+        #[cfg(feature = "filesystem_watcher")] watch_for_changes: bool,
```

Comment:
Please unconditionally add this param. You can panic when it is set without filesystem_watcher enabled though. Cargo features are additive and thus enabling them must not break other crates that don't enable it.

Reply:
I believe I have fixed this. Could you double-check that the issue you pointed out looks good to you now?

Reply:
Looks good now.